src/main/java/Functions.java

```diff
@@ -0,0 +1,142 @@
+        try {
+            int taskNumIndex = Integer.parseInt(input.substring(input.indexOf(' ') + 1)) - 1;
+            if (taskNumIndex >= 0 && taskNumIndex < taskCounter) {
+                if (input.startsWith("mark ")) {
+                    taskList[taskNumIndex].setDone();
+                    System.out.println(SEPARATOR);
+                    System.out.println("Nice! I've marked this task as done:");
+                } else {
+                    taskList[taskNumIndex].setNotDone();
+                    System.out.println(SEPARATOR);
+                    System.out.println("OK! I've marked this task as not done yet:");
+                }
+                System.out.println(taskStatus(taskNumIndex));
+                System.out.println(SEPARATOR);
+            } else {
+                System.out.println(SEPARATOR);
+                System.out.println("Invalid task input. Please try again.");
+                System.out.println(SEPARATOR);
+            }
+        } catch (NumberFormatException e) {
+            System.out.println(SEPARATOR);
+            System.out.println("Invalid task input. Please try again.");
+            System.out.println("Correct format: mark <int>");
+            System.out.println(SEPARATOR);
+        }
+    }
+
+    public String taskStatus(int index){
+        // N tag below in [N] is for no-tag (default setting)
+        return taskList[index].getTaskStatus();
+    }
+
+    private void listTasks() {
+        System.out.println(SEPARATOR);
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskCounter; i++) {
+            System.out.println((i + 1) + ". " + taskStatus(i));
+        }
+        System.out.println(SEPARATOR);
+    }
+
+    private void addTask(String input) {
+        System.out.println(SEPARATOR);
+        if (input.startsWith("todo ")){
+            taskList[taskCounter++] = new ToDos(input);
+        }
+        else if (input.startsWith("deadline ")){
+            taskList[taskCounter++] = new Deadlines(input);
+        }
+        else if (input.startsWith("event ")){
+            taskList[taskCounter++] = new Events(input);
+        }
+        else{
+            taskList[taskCounter++] = new Task(input);
+        }
+
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + taskStatus(taskCounter - 1));
+        System.out.println("Now you have %d task(s) in the list".formatted(taskCounter));
+        System.out.println(SEPARATOR);
+    }
+
+    private void echo(){
+        while(true){
+
+            System.out.println("Say anything! If you are no longer bored, type exit !");
+            String input = in.nextLine();
+            if(input.equals("exit")){
+                System.out.println(SEPARATOR);
+                printFunctions();
+                break;
+            }
+            System.out.println(SEPARATOR);
+            System.out.println(input);
+            System.out.println(SEPARATOR);
```

Comment:
Should System.out.println(SEPARATOR) be its own method, since you have it declared as private final in two different classes?